pyproject.toml

```diff
@@ -9,6 +9,7 @@ requires = [
     # wheels on PyPI
     #
     # see: https://github.com/scipy/oldest-supported-numpy/blob/main/setup.cfg
+    "numpy==1.22.3; python_version=='3.10' and platform_system=='Windows' and platform_python_implementation != 'PyPy'",
     "oldest-supported-numpy",
```

Comment:
The marker `not` does not exist in the grammar defined by the PEP508.
I am wondering if changing the order would be enough then:
- if we have the first match then it will install NumPy 1.22.3
- otherwise, it will default to the oldest compatible NumPy

@ogrisel do you know if the ordering has an impact?

Reply:
[Disjunctive Normal Form](https://en.wikipedia.org/wiki/Disjunctive_normal_form) for the win!